src/test/common/test_crc32c.cc

```diff
@@ -265,3 +265,28 @@ TEST(Crc32c, RangeNull) {
     ASSERT_EQ(crc, *check);
   }
 }
+
+TEST(Crc32c, zeros_performance) {
+  utime_t start;
+  utime_t end;
+
+  for (size_t scale=1; scale < 31; scale++)
+  {
+    size_t size = (1<<scale) + rand()%(1<<scale);
+    start = ceph_clock_now();
+    uint32_t crc_a = ceph_crc32c(111, nullptr, size);
+    end = ceph_clock_now();
+    std::cout << "regular  method. size=" << size << " time= " << (double)(end-start)
+        << " at " << (double)size/(1024*1024)/(double)(end-start) << " MB/sec" << std::endl;
+
+    start = ceph_clock_now();
+    uint32_t crc_b = ceph_crc32c_sctp(111, nullptr, size);
```

Comment:
I think the difference is that we should be comparing the speed of the new algorithm vs the Intel fastzero implementation for all block sizes.  @kestrels test was calling these two against each other:
        crc = ceph_crc32c_intel_fast(crc, (unsigned char const *) NULL, nzerobytes);
vs
        crc = ceph_crc32c_zeros(crc, nzerobytes);

in a main and run with 'time' with these results:
time ./main -f init -z 1073741824 -d
## This is Intel fastzero
crc=0x058E3FA2

real    0m0.046s
user    0m0.044s
sys     0m0.000s
time ./main -f init -z 1073741824 -m
# This is the new PR
crc=0x058E3FA2

real    0m0.096s
user    0m0.092s
sys     0m0.000s

On a processor like this:
processor       : 7
vendor_id       : GenuineIntel
cpu family      : 6
model           : 60
model name      : Intel(R) Xeon(R) CPU E3-1271 v3 @ 3.60GHz